### PR TITLE
[query] add db URL parameter for support specify the database

### PR DESCRIPTION
**Pull request by Junnplus**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add db URL parameter for support specify the database, like: 
`curl --location --request POST 'localhost:8001/v1/statement/?db=system'`

## Changelog

- Improvement

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/2575

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
The `Unit Tests` failed due to the grcov crate, if your want it passed(fixed in https://github.com/datafuselabs/databend/pull/2588), please merge your branch with main and push again :)

**Comment by youngsofun** [comment]
@Junnplus sorry for delay

we are considering redesign api

https://github.com/datafuselabs/databend/issues/2604

File: query/src/servers/http/v1/statement.rs [hunk 1]
```diff
@@ -101,17 +104,21 @@ struct HttpQueryState {
...
             sql,
             state: None,
+            db,
         }
     }
 
     async fn start(&mut self, session_manager: SessionManagerRef) -> Result<HttpQueryState> {
         let session = session_manager.create_session("http-statement")?;
         let ctx = session.create_context().await?;
+        if self.db.is_some() && !self.db.clone().unwrap().is_empty() {
```
Comment (youngsofun) [inline query/src/servers/http/v1/statement.rs:119]: is_empty()  should not be checked specially,  just let set_current_database("")  return Err?

**Comment by youngsofun** [comment]
/lgtm

all others looks good to me.

File: query/src/servers/http/v1/statement.rs (hunk 1 above)
Comment (Junnplus) [reply query/src/servers/http/v1/statement.rs:119]: `/v1/statement/?db=` should be acceptd for default database.

**Pull request merged**